### PR TITLE
Init global babel config

### PR DESCRIPTION
**Pull request by skjnldsv**
Files: https://github.com/search?q=org%3Anextcloud+filename%3Ababel.config.js&type=Code

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+	plugins: [
+		'add-module-exports',
+		'@babel/plugin-syntax-dynamic-import',
+		['@babel/plugin-proposal-class-properties', { loose: true }],
```
Comment (skjnldsv) [inline index.js:5]: So, the discussion is mostly about this.
I think we need to define the defaults.

- I've used `@babel/plugin-proposal-class-properties` on some places, but in the end this causes reactivity with vue, so I'm wondering if that is really worth it.
- `@babel/plugin-syntax-dynamic-import` is really useful to split bundles and async load them
- @PVince81 is `add-module-exports` necessary for all?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  "main": "index.js",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/nextcloud/babel-config.git"
+  },
+  "peerDependencies": {
+    "@babel/core": "^7.13.10",
+    "@babel/plugin-syntax-dynamic-import": "^7.8.3",
+    "@babel/preset-env": "^7.13.12",
+    "babel-plugin-add-module-exports": "^1.0.4"
```
Comment (skjnldsv) [inline package.json:14]: I don't think we should have more than this.
Anything related to eslint like `babel-eslint` is for https://github.com/nextcloud/eslint-config

Any other very essential plugins?

**Comment by skjnldsv** [comment]
@ChristophWurst you use a different one in mail: https://github.com/nextcloud/mail/blob/72c4a44b522bccdf7c6c96102fe7fc7b175a6b82/babel.config.js
Do you have insights related to this?

**Comment by skjnldsv** [comment]
@juliushaertl You use some different plugins on text, any specifics info we should discuss? 
https://github.com/nextcloud/text/blob/009821b5d2e5363f32cdf1594571170cb3cb41a2/babel.config.js

File: index.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+	],
+	presets: [
+		[
+			'@babel/preset-env',
+			{
+				modules: false,
+				useBuiltIns: false,
+				modules: 'commonjs',
+				debug: true,
```
Comment (skjnldsv) [inline index.js:14]: TODO: remove

**Comment by juliushaertl** [comment]
I also needed https://babeljs.io/docs/en/babel-plugin-transform-modules-commonjs in text and https://github.com/nextcloud/richdocuments/pull/1448#issuecomment-809983089 though I'm not sure why this is only an issue with those two apps yet and not with others, so I need to look into this further.

**Comment by skjnldsv** [comment]
> I also needed [babeljs.io/docs/en/babel-plugin-transform-modules-commonjs](https://babeljs.io/docs/en/babel-plugin-transform-modules-commonjs) in text and [nextcloud/richdocuments#1448 (comment)](https://github.com/nextcloud/richdocuments/pull/1448#issuecomment-809983089) though I'm not sure why this is only an issue with those two apps yet and not with others, so I need to look into this further.

Yes, I think we should have this one by default, i'll remov the add-export-module as well as it's not standard.

`@babel/plugin-transform-classes` is needed for private classes vars, but it should not be global

**Comment by skjnldsv** [comment]
Good to go, this includes the most commonly used configs and plugins on nextcloud.
- Commonjs base
- Class syntax
- Dynamic import

**Pull request merged**